src/backend/command.hpp

```diff
@@ -55,6 +58,58 @@ namespace big
 			return m_num_args;
 		}
 
+		virtual std::optional<std::vector<std::string>> get_argument_suggestions(int arg)
+		{
+			return std::nullopt;
+		};
+
+		inline std::optional<int> get_argument_proxy_value(const std::string proxy)
+		{
+			std::string local_player_name_lower = g_player_service->get_self()->get_name();
+			std::string proxy_lower             = proxy;
+			string::operations::to_lower(local_player_name_lower);
+			string::operations::to_lower(proxy_lower);
+
+			// Unfortunately, C++ does not consider std::string as a type and does not have a comparator for it.
+			// So can't really make this a switch case
```

Comment:
Since the comparison is against a single character you could do the following:

```c++
std::string proxy_lower = "#";

// as a single character can be represented numerically you can use a switch case
switch (proxy_lower[0])
{
case '@':
{
    return g_player_service->get_selected()->id();
}
case '!':
{
    return g_player_service->get_closest(true)->id();
}
case '#':
{
    // ....
    return closest->id();
}
}
```

Reply:
Yea, good idea actually!